lib/fs-hash-diff.js

```diff
@@ -0,0 +1,77 @@
+/// @ts-check
+"use strict";
+
+// Imported for type annotations.
+const Entry = require("fs-tree-diff/lib/entry").default; // jshint ignore:line
+const FSTree = require("fs-tree-diff");
+
+class HashEntry {
+  /**
+   * @param relativePath {string}
+   * @param hash {string}
+   */
+  constructor(relativePath, hash) {
+    this.relativePath = relativePath;
+    this.hash = hash;
+  }
+
+  isDirectory() {
+    return false;
+  }
+
+  /**
+   * Whether the entries have the same content.
+   * @param other {unknown}
+   * @returns
+   */
+  equals(other) {
+    if (other instanceof HashEntry) {
+      return this.hash === other.hash;
```

Comment:
`this.hash` can be an empty string. This occurs if while getting a stat there was an error. Is it safe to treat all `HashEntry` created as a result of an error during stating as equal?